app/api/teaching_periods.rb

```diff
@@ -0,0 +1,124 @@
+      else
+        result
+      end
+    end
+
+    desc "Get a teaching period's details"
+    get '/teaching_periods/:id' do
+      teaching_period = TeachingPeriod.find(params[:id])
+      unless (authorise? current_user, User, :get_teaching_periods) || (authorise? current_user, User, :handle_teaching_period)
+        error!({ error: "Couldn't find Teaching Period with id=#{params[:id]}" }, 403)
+      end
+      teaching_period
+    end
+
+    desc 'Update teaching period'
+    params do
+      requires :id, type: Integer, desc: 'The teaching period id to update'
+      requires :teaching_period, type: Hash do
+        optional :period, type: String, desc: 'The name of the teaching period'
+        optional :year, type: Integer, desc: 'The year of the teaching period'
+        optional :start_date, type: Date, desc: 'The start date of the teaching period'
+        optional :end_date, type: Date, desc: 'The end date of the teaching period'
+      end
+    end
+    put '/teaching_periods/:id' do
+      teaching_period = TeachingPeriod.find(params[:id])
+      unless authorise? current_user, User, :handle_teaching_period
+        error!({ error: 'Not authorised to update a teaching period' }, 403)
+      end
+      teaching_period_parameters = ActionController::Parameters.new(params)
+                                                               .require(:teaching_period)
+                                                               .permit(:period,
+                                                                       :year,
+                                                                       :start_date,
+                                                                       :end_date)
+
+      teaching_period.update!(teaching_period_parameters)
+      teaching_period
+    end
+
+    desc 'Get all the Teaching Periods'
+    get '/teaching_periods' do
+      unless authorise? current_user, User, :get_teaching_periods
+        error!({ error: 'Not authorised to get teaching periods' }, 403)
+      end
+      teaching_periods = TeachingPeriod.all
+      result = teaching_periods.map do |c|
+        {
+          id: c.id,
+          period: c.period,
+          year: c.year,
+          start_date: c.start_date,
+          end_date: c.end_date
+        }
+      end
+      result
+    end
+
+    desc 'Delete a teaching period'
+    delete '/teaching_periods/:teaching_period_id' do
+      unless authorise? current_user, User, :handle_teaching_period
+        error!({ error: 'Not authorised to delete a teaching period' }, 403)
+      end
+
+      teaching_period_id = params[:teaching_period_id]
+      TeachingPeriod.find(teaching_period_id).destroy
+    end
+
+    # Returns the current units with teaching periods
+    # Return:
+    # Unit Name, unit ID, teaching period name, year
+    #
+    desc 'Return all the units teaching period information'
+    get '/teaching_periods/id/units' do
```

Comment:
What is this for?

I can imagine getting all of the unit for a given teaching period... but not all with a teaching period set at all...

Reply:
This is for the e-solution project, it is basically returning the units that are associated with a teaching period.

Reply:
Ok - then it needs to be un-authenticated. As this will be accessed without a user login.

Reply:
Is that the data they need? Can you check that?

Reply:
Jake asked me to add this, I will confirm from him when he comes back.